pkg/util/util_test.go

```diff
@@ -560,3 +561,85 @@ func TestCheckPathExists(t *testing.T) {
 		}
 	}
 }
+
+func TestGetIgnoreRulesFromDirectory(t *testing.T) {
+	testDir, err := ioutil.TempDir(os.TempDir(), "odo-tests")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(testDir)
+	tests := []struct {
+		name          string
+		directoryName string
+		fileName      string
+		rulesOnFile   string
+		wantRules     []string
+		wantErr       bool
+	}{
+		{
```

Comment:
all the tests here are testing either `odoignore` or `gitignore`, but there is no case where both files are present.

Reply:
we also need a test case that negates the pattern using `!`